src/cargo/ops/cargo_test.rs

```diff
@@ -91,6 +92,9 @@ fn run_unit_tests(options: &TestOptions,
         };
         let mut cmd = compilation.target_process(exe, pkg)?;
         cmd.args(test_args);
+        if let Some(arg0) = env::args_os().next() {
```

Comment:
If you follow the definition of `target_process` I think you'll find the `fill_env` function which is likely where this should go.